### PR TITLE
Allow ignoring NotFound errors for orphaned pods when draining node

### PR DESCRIPTION
**Pull request by soltysh**
Currently `kubectl drain` fails draining node with orphaned with:
```
error: replicationcontrollers "hello-1" not found: hello-1-f167z, hello-1-f167z
```
This happens when user creates a controller and then removes it with `--cascade=false`, which will remove only the controller, but leaves the pod behind. 

This PR adds `--ingore-not-found` flag to `kubectl drain` commands which allows bypassing those errors.

@mml @kubernetes/sig-cli-pr-reviews ptal
@jupierce @mwoodson fyi

**Release note**:
```release-note
Allow ingoring NotFound errors for orphaned pods when draining node
```

### CODE REVIEW

**Comment by soltysh** [comment]
@sttts @ncdc I wonder if we want to pick it up in the rebase, this error came from a cluster upgrade

**Comment by soltysh** [comment]
@k8s-bot gce etcd3 e2e test this

File: pkg/kubectl/cmd/drain.go [hunk 1]
```diff
@@ -184,6 +185,7 @@ func NewCmdDrain(f cmdutil.Factory, out, errOut io.Writer) *cobra.Command {
 		},
 	}
 	cmd.Flags().BoolVar(&options.Force, "force", false, "Continue even if there are pods not managed by a ReplicationController, ReplicaSet, Job, DaemonSet or StatefulSet.")
+	cmd.Flags().BoolVar(&options.IgnoreNotFound, "ignore-not-found", false, "Ignore \"resource not found\" error when no matching controller is found.")
```
Comment (xiangpengzhao) [inline pkg/kubectl/cmd/drain.go]: Suppose there is a pod managed by Daemonsets controller but the controller is gone for any reason. What if users specify `ignore-not-found` and `ignore-daemonsets` ?

Comment (soltysh) [inline pkg/kubectl/cmd/drain.go]: The most reasonable seems to ignore pods managed by DaemonSets, if such a flag is set. Looking once again at the code, I think I should just re-use `--force` flag in `getPodCreator`, this way what I said above will make sense and will follow what's going on in the other parts of the code. I'll update the PR accordingly.

**Comment by soltysh** [comment]
On second try, I'm wondering how I missed the `--force` flag previously. Maybe because I was looking at 1.5 version in openshift, where the root cause was. With `--force` this is working as expected, thus I'm closing this PR, since it's not needed. 

@jupierce @mwoodson we should get this behavior for 1.6 rebase.

**Pull request closed without merging**